lms/djangoapps/courseware/views/views.py

```diff
@@ -1613,11 +1619,14 @@ def _track_successful_certificate_generation(user_id, course_id):
 @require_http_methods(["GET", "POST"])
 @ensure_valid_usage_key
 @xframe_options_exempt
+@ensure_csrf_cookie
```

Comment:
Are we sure that this change is okay? I'm not sure how exactly to test it in prod liker scenarios, but I could see us changing the behavior of render_xblock having some downstream effects. Like maybe in the LTI domain? I would recommend reaching out to an ancient in platform (Cale, Ormsbee, Nimisha, etc.) who might be able to advise on this.

Reply:
good catch, ill reach out, thanks for this

Reply:
No, injecting the csrf cookie should be ok.